File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    open = 0
    
    for i in s:
        if i == "(":
            open += 1
        elif i == ")":
            if open == 0:
                answer = False
                break
            else:
                open -= 1

    if open > 0:
        answer = False
        
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    let open = 0;
    
    for (let i = 0; i < s.length; i++) {
        if (s[i] === "(") {
            open++
        } else {
            open--
            if (open < 0) {
                answer = false;
                break
            }
        }
    }
    if (open > 0) answer = false

    return answer;
}